ntp-proto/src/time_types.rs

```diff
@@ -201,15 +250,15 @@ ntp_duration_scalar_mul!(isize);
 ntp_duration_scalar_mul!(u8);
 ntp_duration_scalar_mul!(u16);
 ntp_duration_scalar_mul!(u32);
-ntp_duration_scalar_mul!(u64);
-ntp_duration_scalar_mul!(usize);
+// u64 and usize deliberately excluded as they can result in overflows
```

Comment:
There are right now no uses for most of these, however I prefer to have these "just work". The intention here is to allow multiplication (and division) with integer types to the extent that that is safe, without the user (or us) having to worry about which exact integer type they hold at that time. Since rust does not do automatic upgrades of integer types even when safe, I don't view this as an antipattern, but rather as the ergonomic thing to do.

Reply:
Users of the library going around multiplying and dividing `NtpDuration`s seems very niche to me.

Reply:
fair enough, but we definitely will be, and there the same holds in my view.

Reply:
And you think all of these 8 integer types will be used within the library?

Reply:
not neccessarily, but given rusts behaviour around integer widening, I am strongly of the opinion that it is just better to have these and provide the full extent of what is intended rather than to have to deal with it on a case by case basis.